core/kazoo_documents-1.0.0/src/kz_device.erl

```diff
@@ -233,3 +235,53 @@ owner_id(DeviceJObj, Default) ->
 -spec set_owner_id(doc(), ne_binary()) -> doc().
 set_owner_id(DeviceJObj, OwnerId) ->
     wh_json:set_value(?KEY_OWNER_ID, OwnerId, DeviceJObj).
+
+
+-spec timezone(doc()) -> api_binary().
+-spec timezone(doc(), Default) -> ne_binary() | Default.
+timezone(Box) ->
+    timezone(Box, 'undefined').
+timezone(Box, Default) ->
+    case wh_json:get_value(?KEY_TIMEZONE, Box) of
+        'undefined'   -> owner_timezone(Box, Default);
+        <<"inherit">> -> owner_timezone(Box, Default);  %% UI-1808
+        TZ -> TZ
+    end.
+
+-spec owner_timezone(doc(), Default) -> ne_binary() | Default.
+-spec owner_timezone(doc(), Default, kzd_user:doc()) -> ne_binary() | Default.
+owner_timezone(Box, Default) ->
+    case owner(Box) of
+        'undefined'   -> account_timezone(Box, Default);
+        <<"inherit">> -> account_timezone(Box, Default);  %% UI-1808
```

Comment:
@fenollp @macpie How can this clause ever match? Did anyone actually test to see if this clause was reachable?